### PR TITLE
Bug/mail services initialization

### PR DESCRIPTION
**Pull request by kalikaneko**

### CODE REVIEW

**Comment by kalikaneko** [comment]
note that this needs to be merged into feature/async api branch

**Comment by kalikaneko** [comment]
@meskio I got this branch correctly launching smtp, incoming and imap (against my new-mail-api branch).
However, the insert of the incoming mail is raising revisionconflict, I have to check carefully where the error is being produced. Give a look to it if you have some spare time during the weekend, I'm not sure if the problem lies within the mailbox handling, or in the end of the processing pipeline.

**Comment by kalikaneko** [comment]
force pushed one more time, rebased against develop

File: src/leap/bitmask/gui/mainwindow.py [hunk 1]
```diff
@@ -1207,9 +1207,13 @@ def _on_user_logged_in(self):
         self._disconnect_login_wait()
 
         user = self._login_widget.get_logged_user()
-        domain = self._providers.get_selected_provider()
-        full_user_id = make_address(user, domain)
-        self._mail_conductor.userid = full_user_id
+        # XXX the widget now gives us the full user id.
+        # this is confusing.
+        #domain = self._providers.get_selected_provider()
+        #full_user_id = make_address(user, domain)
```
Comment (meskio) [inline src/leap/bitmask/gui/mainwindow.py]: Commented code. Do we really need to keep it here?

File: src/leap/bitmask/services/mail/imap.py [hunk 2]
```diff
@@ -63,9 +63,13 @@ def start_imap_service(*args, **kwargs):
     from leap.bitmask.config import flags
     logger.debug('Launching imap service')
 
+    # TODO --------------------------------------------
+    # this does not make sense in imap anymore, should
+    # be moved to incoming_mail_service
     override_period = get_mail_check_period()
     if override_period:
         kwargs['check_period'] = override_period
+    # -------------------------------------------------
```
Comment (meskio) [inline src/leap/bitmask/services/mail/imap.py]: This has to be moved to `start_incoming_mail_service` my bad. I'll pull-req on this pull-req to fix it.

File: src/leap/bitmask/gui/mainwindow.py (hunk 1 above)
Comment (kalikaneko) [inline src/leap/bitmask/gui/mainwindow.py]: I expect the backend to fix this regression and go back to the older code. I think the full_user_id denomination is clearer than the current behaviour, which I find confusing.

**Comment by meskio** [comment]
Merged and squashed.

**Pull request closed without merging**